clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -292,6 +292,7 @@ private void retry(final long currentTimeMs) {
 
         private void onSuccess(final long currentTimeMs,
                                final OffsetFetchResponse response) {
+            requestState.onSuccessfulAttempt(currentTimeMs);
```

Comment:
Similarly - Request State doesn't matter if the futures are completed. There's one case, where we get an UNSTABLE_OFFSET_COMMIT the same retry function is invoked.